pkg/volume/hostpath/host_path.go

```diff
@@ -489,11 +489,13 @@ func makeDir(pathname string) error {
 // If pathname already exists, whether a file or directory, no error is returned.
 func makeFile(pathname string) error {
 	f, err := os.OpenFile(pathname, os.O_CREATE, os.FileMode(0644))
-	defer f.Close()
 	if err != nil {
 		if !os.IsExist(err) {
 			return err
 		}
 	}
+	if f != nil {
```

Comment:
Move this up before error check (as long as f is not nil, regardless of error call close).